pynecone/state.py

```diff
@@ -61,6 +61,40 @@ def __init__(self, *args, **kwargs):
         for substate in self.get_substates():
             self.substates[substate.get_name()] = substate().set(parent_state=self)
 
+        self._init_mutable_fields()
+
+    def _init_mutable_fields(self):
+        """Initialize mutable fields.
+
+        So that mutation to them can be detected by the app:
+        * list
+        """
+        for field in self.base_vars.values():
+            value = getattr(self, field.name)
+
+            value_in_pc_data = _convert_mutable_datatypes(
+                value, self._reassign_field, field.name
```

Comment:
I found sth interesting. The previous way where we use `lambda` to bind the `field.name` to `self._reassign_field` wouldn't work. The `field.name` would be dynamically evaluated to be the last `field.name` instead of the `field.name` in the current iteration.

So let's say you have 3 fields in a state and the last `state.name` is `friends_in_nested_list` (i.e. the case in the unit test), the `field_name` would always be `friends_in_nested_list`. 😓 I have no choice but to pass both into the class, instead of use `lambda` to piece them together. 

Any ideas for this problem?

Reply:
Ahh I think @Lendemor was seeing a similar issue in handling the dynamic routing - I think this code may be relevant where he used a factory? https://github.com/pynecone-io/pynecone/blob/5aae6a122d8935aea599b5a9f8bf84622fda9cdb/pynecone/state.py#L291

I'm not entirely sure though may have to give this a deeper look.